core/src/main/java/hudson/scm/SCM.java

```diff
@@ -522,13 +522,22 @@ public void postCheckout(AbstractBuild<?,?> build, Launcher launcher, FilePath w
      * (for example, SVN revision number.)
      *
      * <p>
-     * This method is invoked whenever someone does {@link AbstractBuild#getEnvironment(TaskListener)}, which
+     * This method is invoked whenever someone does {@link Run#getEnvironment(TaskListener)}, which
      * can be before/after your checkout method is invoked. So if you are going to provide information about
      * check out (like SVN revision number that was checked out), be prepared for the possibility that the
      * check out hasn't happened yet.
      */
-    // TODO is an equivalent for Run needed?
+    public void buildEnvVars(@Nonnull Run<?,?> build, @Nonnull Map<String,String> env) {
```

Comment:
It is rather painful and error-prone to have overloads which differ only in supertype/subtype relationship. You can only distinguish them using explicit casts.

While this can be solved by reordering parameters, I think it is better to pick a new name for the new overload, such as `buildEnvironment`.